### ISSUE TITLE
BlendTree Nodes invalid just right after added.

### ISSUE THREAD

**Issue by Angeloss** [godotengine/godot#24088]
**Godot version: 5b8dde4b4a5a930e08da51e07ec87f43e5bd7b36**

**OS/device including version:**
Windows 10

**Issue description:**
Any new node renamed automatically by the BlendTree editor with a consecutive number ("Animation 2" "Blend2 2") is invalid.

`ERROR: AnimationNodeBlendTree::get_node: Condition ' !nodes.has(p_name) ' is true. returned: Ref<AnimationNode>()
At: scene\animation\animation_blend_tree.cpp:884`

`ERROR: AnimationNodeBlendTreeEditor::_node_selected: Condition ' !anode.is_valid() ' is true.
At: editor\plugins\animation_blend_tree_editor_plugin.cpp:428`

**Steps to reproduce:**
In an AnimationNodeBlendTree, add an animation node, then add a second, this new "Animation 2"  node is invalid.

![notvalid](https://user-images.githubusercontent.com/862976/49305644-66650100-f495-11e8-9265-d5cf0c68b96b.jpg)

**Comment by Clean3d** [comment]
I've experienced this bug on Ubuntu14.04 and Windows 8. I'm new to engine debugging, but this issue seems to appear in this commit:
https://github.com/godotengine/godot/commit/1025e3ecea67b90232cfdc9590a7ee8887efdcc8

Debugging in Visual Studio Community, it appears the modified code strips the space character when testing whether the output node is valid(?). For instance, in scene/animation/animation_blend_tree.cpp...

`AnimationNodeBlendTree::ConnectionError AnimationNodeBlendTree::can_connect_node(const StringName &p_input_node, int p_input_index, const StringName &p_output_node) const {

	if (!nodes.has(p_output_node) || p_output_node == SceneStringNames::get_singleton()->output) {
		return CONNECTION_ERROR_NO_OUTPUT;
	}`

...`!nodes.has(p_output_node)` returns false. The debugger says p_input_node == "Animation2" in the broken commit, and "Animation 2" in the previous commit.

**Comment by Clean3d** [comment]
I've made a little progress, I think... Inside AnimationBlendTreeEditor::_add_node() there's some code that iterates through available node names until it finds one that is not taken (e.g. if "Animation" is present, try "Animation 2", etc). The code that creates that name-string doesn't take in to account the selected name-number separator from Project Settings>>Node>>Name Num Separator. I changed it to use _get_name_num_separator() instead of " ", and nodes created with a name-num-separator OTHER THAN " " seem to work.

At some point, space-separators are either removed or ignored when referring to one of these problem-nodes.

(...or I could be wrong about everything. :stuck_out_tongue: )
